client/state/sites/selectors.js

```diff
@@ -985,3 +985,19 @@ export function getCustomizerUrl( state, siteId ) {
 		'return': returnUrl
 	}, adminUrl );
 }
+
+/*
+ * Returns true if the site has unchanged site title
+ *
+ * @param {Object} state Global state tree
+ * @param {Object} siteId Site ID
+ * @return {Boolean} True if site title is default, false otherwise.
+ */
+export const hasDefaultSiteTitle = ( state, siteId ) => {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+	const slug = getSiteSlug( state, siteId );
+	return site.name === i18n.translate( 'Site Title' ) || slug.indexOf( site.name ) === 0;
```

Comment:
Arguably, the mindset when reviewing code is different from that when reading committed code. What I mean by this is that, when reviewing, you're more likely to think things through and try to poke holes in the logic, while when the code is there (and assumed to be working) you're more likely to fill the gaps in your understanding by inferring what makes sense from the logic and the naming of things. In other words, upon reading the name `hasDefaultSiteTitle` and looking at the overall shape of the return value (`return someCheck( name ) || someOtherCheck( name, slug )`) one can figure out the writer's intention.

I'm not saying it's necessarily _not_ confusing or that it doesn't require an explanatory comment. I'm just giving food for thought, to potentially avoid comments. :)